### ISSUE TITLE
Geolocator.getPositionStream distanceFilter not working on web

### ISSUE THREAD

**Issue by BoHellgren** [Baseflow/flutter-geolocator#674]
## 🐛 Bug Report

I run this code
```
Geolocator.getPositionStream(desiredAccuracy: LocationAccuracy.high, distanceFilter: 5)
            .listen((Position position) {
          if (position.accuracy < 25) showPos(position.latitude, position.longitude);
        });
```
### Expected behavior
If the device is not moving, the callback should not be executed more than once, since distanceFilter >0.
But on the web, it is executed repeatedly, about once per second.

### Configuration

**Version:** 6.2.1

Flutter 1.26.0-17.6.pre • channel beta • https://github.com/flutter/flutter.git
Framework • revision a29104a69b (7 days ago) • 2021-02-16 09:26:56 -0800
Engine • revision 21fa8bb99e
Tools • Dart 2.12.0 (build 2.12.0-259.12.beta)

**Comment by mvanbeusekom** [comment]
@BoHellgren unfortunately `distanceFilter` is not supported on the web platform. The web API doesn't have any filter based on distance so the parameter is simply ignored. 

I have marked this method "up for graps" as I think it should be relatively easy to implement a custom solution for this. Since the web API ([official documentation](https://api.dart.dev/stable/2.10.5/dart-html/Geolocation/watchPosition.html)) already returns a `Stream<Geoposition>` we should be able to filter it with a method like `skipWhile` and skip positions as long as they don't match the distance criteria. To calculate the distance between two locations we can use the `distanceBetween` method which is implementatie in the `GeolocatorPlatform` base class.

I was thinking something along the lines of:

```dart
  @override
  Stream<Position> getPositionStream({
    LocationAccuracy desiredAccuracy = LocationAccuracy.best,
    int distanceFilter = 0,
    bool forceAndroidLocationManager = false,
    int timeInterval = 0,
    Duration? timeLimit,
  }) {
    if (!_locationServicesEnabled) {
      throw LocationServiceDisabledException();
    }

    return _geolocation!
        .watchPosition(
          enableHighAccuracy: _enableHighAccuracy(desiredAccuracy),
          timeout: timeLimit,
        )
        .handleError((error) => throw _convertPositionError(error))
        .skipWhile((geoposition) {
          // If no distance filter supplied, don't skip
          if (distanceFilter == 0) {
            return false;
          }
          
          // TODO: Compare with earlier position and add filter logic
        })
        .map(_toPosition);
  }
```

**Comment by BoHellgren** [comment]
Thanks. The fact that distanceFilter is not supported on the web platform should be documented in the README. That would have saved me, and may save other developers, quite a bit of time. 

I made my own distanceFilter using _Geolocator.distanceBetween_, as you suggested. My code is below, for anyone interested. Most of it works in most environments. But there are additional bugs in the permission handling on the web. I realize that the multitude of environments to be supported (iOS various releases, Android various releases, Chrome on Windows, Chrome on Android, Chrome on iOS, Safari, ...) makes it difficult to test this well, and I have seen that you will change this in version 7, so I will probably not bother to create any more Issues.

```
void watchPos() async {
    bool isLocationEnabled = await Geolocator.isLocationServiceEnabled();
    if (isLocationEnabled) {
      LocationPermission geolocationPermission = await Geolocator.checkPermission();
      print('********geolocationPermisssion 1 $geolocationPermission');
      if (geolocationPermission != LocationPermission.deniedForever) {
        if (geolocationPermission == LocationPermission.denied)
          geolocationPermission = await Geolocator.requestPermission();
        print('********geolocationPermisssion 2 $geolocationPermission');
        if (geolocationPermission == LocationPermission.always ||
            geolocationPermission == LocationPermission.whileInUse) {
          print('********geolocationPermisssion 3 $geolocationPermission');
          Geolocator.getPositionStream(desiredAccuracy: LocationAccuracy.high, distanceFilter: 5)
              .listen((Position position) => newPosition(position));
          return;
        }
      }
    }
    WidgetsBinding.instance
        .addPostFrameCallback((_) => MyApp.rootScaffoldMessengerKey.currentState.showSnackBar(SnackBar(
              content: Text(
                  "Platstjänster ej aktiverade, eller behörighet saknas.\nAppen kan inte visa din position."),
              duration: Duration(seconds: 10),
            )));
  }

  void newPosition(Position position) async {
    print('********************new position. Accuracy ${position.accuracy}');
    if (position.accuracy > 25) return;

    if (kIsWeb) {
      // Fix for issue https://github.com/Baseflow/flutter-geolocator/issues/674
      Marker previous = _markers.firstWhere((m) => m.markerId.value == 'user', orElse: () => null);
      if (previous == null) {
        _markers.add(Marker(
            markerId: MarkerId('user'),
            position: LatLng(position.latitude, position.longitude),
            icon: BitmapDescriptor.defaultMarker,
            infoWindow: InfoWindow(title: 'Här är du nu')));
        setState(() {});
        return;
      }
      double distMoved = Geolocator.distanceBetween(
          position.latitude, position.longitude, previous.position.latitude, previous.position.longitude);
      print('************distMoved $distMoved');
      if (distMoved < 5) return;
    } // end of kIsWeb

    _markers.removeWhere((m) => m.markerId.value == 'user');
    _markers.add(Marker(
        markerId: MarkerId('user'),
        position: LatLng(position.latitude, position.longitude),
        icon: BitmapDescriptor.defaultMarker,
        infoWindow: InfoWindow(title: 'Här är du nu')));

    if (kIsWeb) {
      // Fix for bug https://github.com/flutter/flutter/issues/65255
      Set<Marker> _markers_save = Set<Marker>.from(_markers);
      _markers.clear();
      setState(() {});
      await Future.delayed(Duration(milliseconds: 250));
      _markers = Set<Marker>.from(_markers_save);
    }

    setState(() {});
  }
```

**Comment by JDDV** [comment]
Hi @BoHellgren, the check to see if the location has been changed using the distanceFilter parameter has been implemented in version ^2.0.2. thank you for bringing this bug to our attention!
